segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/extendeds3/ExtendedS3Storage.java

```diff
@@ -263,6 +264,12 @@ private boolean syncExists(String streamSegmentName) {
     private SegmentProperties syncCreate(String streamSegmentName) throws StreamSegmentExistsException {
         long traceId = LoggerHelpers.traceEnter(log, "create", streamSegmentName);
 
+        /* To ensure put-if-absent semantics we need to use If-None-Match tag as described here:
+         * http://www.emc.com/techpubs/api/ecs/v3-0-0-0/S3ObjectOperations_createOrUpdateObject_7916bd6f789d0ae0ff39961c0e660d00_ba672412ac371bb6cf4e69291344510e_detail.htm
+         * But it does not work. As a work around If-Unmodified-Since tag is used.
+         * If the object is modified (created or created and written to) since the listObject check, the putObject call fails.
+         */
+        Date creationTime = new Date();
```

Comment:
I'm not sure this works, it seems to require the clock of the server running this and the ECS one to be synchronized because `setIfUnmodifiedSince` below will compare the given timestamp to the modification time in ECS. Even if we assume synchronized clocks, it is possible that this server gets the date another pauses (say long GC cycle), while another server writes the object. Such situations would cause the problem we are trying to solve.

I think this workaround is not doing much good. We should instead just leave the create as is, with the known bug, and add a TODO describing the problem, with an issue number, which we will fix as soon as the extended s3 lib we are using get's fixed.